src/Core/Domain/Product/Combination/Query/GetProductCombinationsForEditing.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Combination\Query;
+
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * Retrieves product combinations
+ */
+class GetProductCombinationsForEditing
+{
+    /**
+     * @var ProductId
+     */
+    private $productId;
+
+    /**
+     * @var LanguageId
+     */
+    private $languageId;
+
+    /**
+     * @var int
+     */
+    private $limit;
+
+    /**
+     * @var int
+     */
+    private $offset;
+
+    /**
+     * @param int $productId
+     * @param int $languageId
+     * @param int $limit
+     * @param int $offset
+     */
+    public function __construct(
+        int $productId,
+        int $languageId,
+        //@todo: where do we put default limit? Or leave it required here and take care of it in presentation? (like in some grid filters)
+        int $limit,
```

Comment:
If we set a default limit here, as I suggested earlier, it would be 0 or null to allow disabling pagination Other than that the default pagination number shouldn't be handled here but rather in the presentation code as you suggested